.github/workflows/release.yml

```diff
@@ -0,0 +1,76 @@
+name: Release
+
+on:
+  workflow_dispatch:
+
+permissions:
+  contents: write
+
+jobs:
+  verify-branch:
+    name: Verify that runs on the main branch
+    runs-on: ubuntu-latest
+    steps:
+      - name: Fail if branch is not main
+        if: github.ref != 'refs/heads/main'
+        run: |
+          echo "The release workflow should only be triggered on the main branch"
+          exit 1
+
+  get-version:
+    name: Get version from Cargo.toml
+    needs: verify-branch
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.version.outputs.version }}
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Get version from Cargo.toml
+        id: lookupVersion
+        uses: mikefarah/yq@f15500b20a1c991c8729870ba60a4dc3524b6a94
+        with:
+          cmd: yq -oy '"v" + .workspace.package.version' 'Cargo.toml'
+
+      - name: Print version
+        id: version
+        run: |
+          VERSION=${{ steps.lookupVersion.outputs.result }}
+          echo "version=$VERSION" >> "$GITHUB_OUTPUT"
+
+  verify-tag-existence:
+    name: Check if version exists as GitHub tag
+    needs: get-version
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Check if version exists as a GitHub tag
+        id: check_tag
+        run: |
+          VERSION=${{ needs.get-version.outputs.version }}
+          if git rev-parse "v$VERSION" >/dev/null 2>&1; then
```

Comment:
You can go one step further, and do something like this:

```
$ curl -# -Ls --fail -H 'Accept: application/json' "https://github.com/software-mansion/cairo-annotations/releases/0.0.1"
```

If given release exists it will give a response that looks similar to this:

```
{"id":123,"tag_name":"0.0.1","update_url":"/software-mansion/cairo-annotations/releases/tag/0.0.1","update_authenticity_token":"<token>",(.....)}
```

and if the version does not exist, you'll just get:

```
$ curl -# -Ls --fail -H 'Accept: application/json' "https://github.com/software-mansion/cairo-annotations/releases/0.0.1"
$ echo $?
22
```

That way you check not only for the tag existence, but also for the release itself

Reply:
Modified to do that